catalog/spatial/registry/registry-admin-modules/registry-admin-remote-ui/src/main/webapp/js/view/Registry.view.js

```diff
@@ -195,14 +196,33 @@ function (ich,Marionette,_,$,Q,ModalRegistry,EmptyView,wreqr,Utils,deleteRegistr
             });
         },
         addRegistry: function(){
-            if(this.model) {
-                wreqr.vent.trigger("showModal",
-                    new ModalRegistry.View({
-                        model: this.model.getRegistryModel(),
-                        registry: this.model,
-                        mode: 'add'
-                    })
-                );
+            var self = this.model;
+            var configNames = self.getRegistryModel().get('configNames');
+            var regConfigs = self.getRegistryModel().get('registryConfiguration');
+            regConfigs.forEach(function(config){
+                var service = config.get('service').get('name');
+                if(configNames.indexOf(service) === -1){
+                    configNames.push(service);
+                }
+            });
```

Comment:
Handlebars can access the config names directly from the registryConfigurations. It would have to dig a bit. If you do want to go the configNames list route, should it be populated in the model instead of at every plus click?